### DISCUSSION TITLE
Transformar vetor clássico em metodos Reduce, Filter e Map. (numero primo)

### DISCUSSION THREAD

**Discussion by AdemilsonSimiao** [frontendbr/forum#2249] [Perguntas e Respostas]
Ola pessoal! Gostaria de saber como passa esse código funcional para os metodos Redux, Filter e Map.

var primo = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 9, 8, 7]
for(numero=2; numero <= primo.length; numero++){
    let nprimo = true;
    for(divisor=2; divisor< numero; divisor++){
        if(numero%divisor ===0){
            nprimo = false;
        }
    }
    if(nprimo)console.log(numero);
}

//Esse foi o que tentei e acabou não dando certo
var fprimo = primo.map(primo =>{
    for(let numero = 2; numero <= primo; numero++){
        let sprimo = true;
        for(let divisor = 2; divisor < numero; divisor++){
            if(numero % divisor === 0)
            sprimo=false;
        }
        if(sprimo);
    }
})
console.log(fprimo)

**Comment by felquis** [comment]
Sugestões:
1. Formate seus trechos de código usando markdown para ajudar a gente a te ajudar
2. Em vez de Redux, vc quis dizer Reduce, se puder alterar tornará o fórum mais bem formatado.

Veja se este artigo consegue te dar uma direção https://desenvolvimentoparaweb.com/javascript/map-filter-reduce-javascript/
